isis/tests/FunctionalTestsLeisa2isis.cpp

```diff
@@ -87,4 +88,297 @@ TEST(leisa2isisTest, leisa2isisTestDefault) {
+
+   ASSERT_DOUBLE_EQ(bandbin["Center"][0].toDouble(), 2.4892);
+   ASSERT_DOUBLE_EQ(bandbin["Center"][64].toDouble(), 1.9784);
+   ASSERT_DOUBLE_EQ(bandbin["Center"][128].toDouble(), 1.572);
+   ASSERT_DOUBLE_EQ(bandbin["Center"][255].toDouble(), 2.0898);
+
+   ASSERT_DOUBLE_EQ(bandbin["Width"][0].toDouble(), 0.011228);
+   ASSERT_DOUBLE_EQ(bandbin["Width"][64].toDouble(), 0.008924);
+   ASSERT_DOUBLE_EQ(bandbin["Width"][128].toDouble(), 0.007091);
+   ASSERT_DOUBLE_EQ(bandbin["Width"][255].toDouble(), 0.004915);
+
+   ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][0].toDouble(), 1);
+   ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][64].toDouble(), 65);
+   ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][128].toDouble(), 129);
+   ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][255].toDouble(), 256);
+ }
+
+
+TEST(leisa2isisTest, leisa2isisTestRawErrormapFail) {
+   Pvl appLog;
+   QTemporaryDir prefix;
+   QString dAPP_XML = FileName("$ISISROOT/bin/xml/leisa2isis.xml").expanded();
+   QString cubeFileName = prefix.path() + "/leisa2isisTEMP.cub";
+   QString errFileName = prefix.path() + "/leisa2isisTEMPerr.cub";
+
+   QVector<QString> args = {"from=data/leisa2isis/raw.fit",
+                            "to=" + cubeFileName,
+                            "errormap=" + errFileName};
+
+   UserInterface options(dAPP_XML, args);
+   // Raw files with an errormap specification should fail
+   ASSERT_ANY_THROW(leisa2isis(options, &appLog));
+ }
+
+
+TEST(leisa2isisTest, leisa2isisTestRawQualityFail) {
+   Pvl appLog;
+   QTemporaryDir prefix;
+   QString dAPP_XML = FileName("$ISISROOT/bin/xml/leisa2isis.xml").expanded();
+   QString cubeFileName = prefix.path() + "/leisa2isisTEMP.cub";
+   QString qualityFileName = prefix.path() + "/leisa2isisTEMPqual.cub";
+
+   QVector<QString> args = {"from=data/leisa2isis/raw.fit",
+                            "to=" + cubeFileName,
+                            "quality=" + qualityFileName};
+
+   UserInterface options(dAPP_XML, args);
+
+   // Raw files with a quality specification should fail
+   ASSERT_ANY_THROW(leisa2isis(options, &appLog));
+}
+
+TEST(leisa2isisTest, leisa2isisTestQualityReplacement) {
+   Pvl appLog;
+   QTemporaryDir prefix;
+   QString cubeFileName = prefix.path() + "/leisa2isisTEMP.cub";
+   QVector<QString> args = {"from=data/leisa2isis/lsb_0034933739_0x53c_sci_1_cropped.fit",
+                            "to=" + cubeFileName,
+                            "replace=true"};
+
+  UserInterface options(APP_XML, args);
+  try {
+   leisa2isis(options, &appLog);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to ingest LEISA image: " << e.toString().toStdString().c_str() << std::endl;
+  }
+  Cube cube(cubeFileName);
+
+  Histogram *hist = cube.histogram();
+
+  ASSERT_DOUBLE_EQ(hist->Average(), 9264735084251.8848);
+  ASSERT_DOUBLE_EQ(hist->Sum(), 7059728134199936);
+  ASSERT_EQ(hist->ValidPixels(), 762);
+  ASSERT_DOUBLE_EQ(hist->StandardDeviation(), 16184539944722.791);
+
```

Comment:
A bonus for this would be to actually check to make sure that one of the pixels that is supposed to be replaced based on the quality map is actually replaced, but I think this is okay as-is.